src/sonic_ax_impl/mibs/ieee802_1ab.py

```diff
@@ -536,18 +544,18 @@ def update_rem_if_mgmt(self, if_oid, if_name):
             return
         self.if_range.sort()
 
-    def _update_per_namespace_data(self, db_conn, pubsub):
```

Comment:
>_update_per_namespace_data [](start = 8, length = 26)

Original code seems better. Why change?

Reply:
Before namespace changes, self.pubsub was getting updated in update_data(). After namespace change, self.pubsub array was not updated as we pass the instance. By passing the array index, we can ensure that the array is directly getting udpated. So this fix is to make sure the code is as it was before namespace changes.

Reply:
Updated description with the issue and why this fix is required.

Reply:
I see the bug in [https://github.com/Azure/sonic-snmpagent/pull/131](https://github.com/Azure/sonic-snmpagent/pull/131).

To make code cleaner, could you try add
```
db_conn = self.db_conn[inst]
pubsub = self.pubsub[inst]
if not pubsub:
    ...
    self.pubsub[inst] = pubsub

```

---
In reply to: [441919551](https://github.com/Azure/sonic-snmpagent/pull/134#discussion_r441919551) [](ancestors = 441919551)

Reply:
Fixed as per comments.